Description: Use SSLv23 method instead of SSLv3
 Taken from upstream revision c225795793696c5b5d4a9243ebeb129091229ca5
Author: ThomasV
Forwarded: yes
Last-Update: 2014-11-23

Index: electrum/lib/interface.py
===================================================================
--- electrum.orig/lib/interface.py	2014-11-23 09:51:02.756184724 +0200
+++ electrum/lib/interface.py	2014-11-23 09:51:53.444984556 +0200
@@ -333,7 +333,7 @@
                         continue
 
                     try:
-                        s = ssl.wrap_socket(s, ssl_version=ssl.PROTOCOL_SSLv3, cert_reqs=ssl.CERT_NONE, ca_certs=None)
+                        s = ssl.wrap_socket(s, ssl_version=ssl.PROTOCOL_SSLv23, cert_reqs=ssl.CERT_NONE, ca_certs=None)
                     except ssl.SSLError, e:
                         print_error("SSL error retrieving SSL certificate:", self.host, e)
                         s = None
@@ -379,7 +379,7 @@
         if self.use_ssl:
             try:
                 s = ssl.wrap_socket(s,
-                                    ssl_version=ssl.PROTOCOL_SSLv3,
+                                    ssl_version=ssl.PROTOCOL_SSLv23,
                                     cert_reqs=ssl.CERT_REQUIRED,
                                     ca_certs= (temporary_path if is_new else cert_path),
                                     do_handshake_on_connect=True)
